lib/diagnostic.js

```diff
@@ -6,23 +6,44 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+const normalizeWords = function (paragraph) {
+  let words = paragraph.split(/\s+/);
+  let normalizedWords = [];
+  for (let i = 0; i < words.length; i++) {
+    let word = words[i].replace(/\W+/, '');
+    normalizedWords[i] = word.toUpperCase();
+  }
+
+  return normalizedWords;
 };
 
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const uniqueWords = function (paragraph) {
+  let normalizedWords = normalizeWords(paragraph);
+  let uniqueWords = {};
+  for (let i = 0; i < normalizedWords.length; i++) {
+    let word = normalizedWords[i];
+    uniqueWords[word] = true;
+  }
+
+  return uniqueWords;
 };
 
 // 3. Write a function that takes a string argument and returns the count of
 // words in a string.
 // Provide the *option* to count unique words instead of total words, by passing
 // a second argument.
 
-const wordCount = function (/* */) {
-  /* your response here */
+const wordCount = function (paragraph, countUnique = true) {
```

Comment:
A parameter of function is not initialized inside the parenthesis. What you intended to do (I think!) was this:
```
const wordCount = function (paragraph, countUnique) {
   // some code in here
}
```